### DISCUSSION TITLE
Allow test methods to be static

### DISCUSSION THREAD

**Discussion by feO2x** [thomhurst/TUnit#1283] [Ideas]
In xunit.net, I can write static test classes with static test methods in them.

```csharp
public static MyTestClass
{
    [Fact]
    public static void MyTest() { }
}
```

While this is not a widely adopted pattern, I like it when I can make tests static as this indicates it does not rely on instance fields.

Using version 0.3.43, TUnit unfortunately does not pick up static methods. Do you think it's possible to add this feature to TUnit?

<img width="868" alt="image" src="https://github.com/user-attachments/assets/c38fdb81-b13c-449e-9cc9-6bbd60ee99c1">

**Comment by thomhurst** [comment]
A lot of the internal code is worked around instance members for tests and so would be a massive refactor without much benefit (IMO). I'd also have to build more complicated source code generation. It would be a real headache.

I also just don't like the idea of static tests. They can more easily lead to leaky state due to static fields/properties/etc. TUnit can prevent that when you use instance members.

I would just adopt the pattern of using instance members :)  It's less code anyway! (albeit by only one keyword.)

**Comment by feO2x** [comment]
That's understandable. Thank you for the clarification and all the efforts you put into TUnit.
